examples/stable-diffusion/training/train_text_to_image_sdxl.py

```diff
@@ -301,7 +260,16 @@ def parse_args(input_args=None):
         default=1024,
         help=(
             "The resolution for input images, all the images in the train/validation dataset will be resized to this"
-            " resolution"
+            " resolution."
+        ),
+    )
+    parser.add_argument(
+        "--crop_resolution",
+        type=int,
+        default=1024,
+        help=(
+            "The resolution for cropping input images, all the images in the train/validation dataset will be resized to this"
+            " resolution."
```

Comment:
we only do crop only if the crop_resolution is less than resolution. See preprocess_trainining

Reply:
Okay. Shouldn't it be `args.crop_resolution` instead of `args.resolution` here
```py
train_crop = transforms.CenterCrop(args.resolution) if args.center_crop else transforms.RandomCrop(args.resolution)
```
?

Reply:
@regisss 
The original code is from [here](https://github.com/huggingface/diffusers/blob/01ac37b331d87368741d5f9554eaf47e0dd05a59/examples/text_to_image/train_text_to_image_sdxl.py#L842).

I think it should have been `args.crop_resolution` in the original as well, but that might be a bug. Should we fix it to `args.crop_resolution` or keep it aligned to original?

Reply:
Yes, I also think it's a mistake in the original example. Let's keep it that way then, I'll ask the Diffusers team about it.

Reply:
So actually it's not a bug. The crops are squared images whereas images are resized with the smaller side equal to `args.resolution` and the other proportionally resized.